internal/api/mfa.go

```diff
@@ -289,6 +289,45 @@ func (a *API) invokeHook(ctx context.Context, input, output any) error {
 		}
 
 		return nil
+	case *hooks.CustomAccessTokenInput:
+		hookOutput, ok := output.(*hooks.CustomAccessTokenOutput)
+		if !ok {
+			panic("output should be *hooks.CustomAccessTokenOutput")
```

Comment:
sorry i can't remember if i asked this previously but why do we opt to `panic` here?

Reply:
Oh I think it's because the output is defined by us - if the typecast fails it indicates there's a programmer error and should be be dealt with via a panic: 

I think the thread is here: https://github.com/supabase/gotrue/pull/1328#discussion_r1413386685 and there's another discussion in the link in the above thread.